lua/entities/ent_jack_40mmgrenade/init.lua

```diff
@@ -44,20 +44,22 @@ function ENT:Detonate()
     self.Exploding = true
     local pos = self:GetPos()
 
-    util.ScreenShake( pos, 99999, 99999, 1, 750 )
-    local attacker = self:GetNWEntity( "Owner" )
+    util.ScreenShake( pos, 200, 20, 1, 750 )
+    util.ScreenShake( pos, 1, 20, 2, 2000 )
+    local owner = self:GetNWEntity( "Owner" )
+    local attacker = owner
     if IsValid( attacker ) then
         local creator = attacker:GetCreator()
         attacker = IsValid( creator ) and creator or attacker
     end
 
-    util.BlastDamage( self, attacker, pos, 190, 190 )
+    util.BlastDamage( self, attacker, pos, owner.BulletDamage, owner.BulletDamage )
 
     local plooie = EffectData()
     plooie:SetOrigin( self:GetPos() )
     plooie:SetScale( .75 )
     plooie:SetRadius( 2 )
-    plooie:SetNormal( self:GetForward() )
+    plooie:SetNormal( Vector( 0,0,1 ) )
```

Comment:
Creating vectors on the fly is slow. Could you assign this to a top level variable? _(also, spaces after commas)_

Reply:
Applied